### ISSUE TITLE
[Feature request] - Add interpolation helper for components

### ISSUE THREAD

**Issue by nicolasigot** [kazupon/vue-i18n#455]
Hello,

The goal of this request is to have helper function available associated to the $i18n instance in the vue application.
This helper is to provide possibility to use the interpolation feature on existing labels.
Currently we have $t('key', [params]) which do it for resources available in the language bundle.
When I would request is to have an other method which will do:
$ti('This is {0} value', ['my']) on which we can provide already localized values as parameter.

In order to do it within my component I currently do:
this.$i18n.formatter.interpolate('This is {0} value', ['my']).join('');

It is working fine but having a quick helper would be great.
Let me know if you need more information.
Thanks,
Nicolas

**Comment by dobromir-hristov** [comment]
Isnt this what `this.$t()` is for?

**Comment by nicolasigot** [comment]
Hello,
Thanks for your feedback. this.$t only works with a key as a parameter which should match a key/value in the language bundle.
What I would like is to have the possibility to pass a formatted value instead of a key as parameter. This will allow us to use the interpolation engine without looking for existing key / value in the language bundle.

As an example, the value I want to pass is coming from a backend request as localized value in which I want to pass a parameter:

var backendData = 'My age is {0} years old';
var myAge = 25

$ti(backendData, [myAge]) => My age is 25 years old

As you can see backendData is already localized value and not a key available in the resource bundle.
Does it make sense ?
Let me know if you need more information.

**Comment by nicolasigot** [comment]
@dobromir-hristov Thanks for the tip. I already have a workaround from my first comment: this.$i18n.formatter.interpolate('This is {0} value', ['my']).join('');
I just requested to have a shortcut as this is for me a real feature I need in many cases and I do not think being the only one ;)

**Comment by dobromir-hristov** [comment]
Yeah, now I totally got you. I think this is something that is not as often used, so I am not sure whether @kazupon will agree on adding a specific helper for it. The method I showed is not as long, maybe you can create your own wrapper around it and attach it to your vue prototype?

**Comment by nicolasigot** [comment]
Thanks @dobromir-hristov . This is exactly what I am sort of doing but I was thinking that it could be useful to add it as a feature via helper. I let @kazupon decide now if this makes sense to be added or not.
